library/src/main/java/com/cube/storm/ui/view/holder/list/ToggleableListItemViewHolder.java

```diff
@@ -95,23 +86,36 @@ public ToggleableListItemViewHolder(View view)
 					eventHook.onViewClicked(itemView, model);
 				}
 
-				if (toggleContainer.getVisibility() == View.GONE)
-				{
-					toggleContainer.setTag(true);
-					toggleContainer.setVisibility(View.VISIBLE);
-					expandIcon.setImageResource(R.drawable.ic_collapse);
-					embeddedLinksContainer.setVisibility(View.VISIBLE);
-					title.setContentDescription(titleText + ". Description expanded");
-				}
-				else
-				{
-					toggleContainer.setTag(false);
-					toggleContainer.setVisibility(View.GONE);
-					expandIcon.setImageResource(R.drawable.ic_expand);
-					embeddedLinksContainer.setVisibility(View.GONE);
-					title.setContentDescription(titleText + ". Description collapsed");
-				}
+				// Invert expansion state when clicked
+				expandContainer(!model.getExpanded(), model);
 			}
 		});
 	}
+
+	/**
+	 * Expands a {@link ToggleableListItem} and retains it's expanded state for
+	 * redrawing the view when scrolling
+	 *
+	 * @param expand true to expand, false to collapse
+	 * @param model the model with which to persist the expanded state
+	 */
+	private void expandContainer(boolean expand, ToggleableListItem model)
+	{
+		toggleContainer.setTag(expand);
+		model.setExpanded(expand);
+		if (expand)
+		{
+			toggleContainer.setVisibility(View.VISIBLE);
+			expandIcon.setImageResource(R.drawable.ic_collapse);
+			embeddedLinksContainer.setVisibility(View.VISIBLE);
+			title.setContentDescription(titleText + ". Description expanded");
```

Comment:
Should these content descriptions be localised?

Reply:
I believe `titleText` is already localised and that this is the same logic as before. However maybe it's something we can look at.